### PR TITLE
Ports - Faiza

### PR DESCRIPTION
**Pull request by Faiza1987**
# JS Adagrams
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
--- | ---
What patterns were you able to use from your Ruby knowledge to apply to JavaScript? | declare variables and assign values to them; iterate over arrays and objects(hashes in Ruby); use console.log to display information for debugging; pass variables into functions to use as parameters.
Did you need to use different strategies to find information online about JavaScript than you would for Ruby? | No, I used similar techniques to what I used in Ruby to find how I would do something in JavaScript (such as deleting something from an array).
What was a challenge you were able to overcome on this assignment? | How to rewrite the methods from the Ruby Adagrams project into JavaScript. Some methods had to be written in a different way because JavaScript does not have the Range data type.
What has been interesting and positive about learning a new programming language? | How easy it was! The concepts were the same but I had to adjust to how functions and classes work in JS v. Ruby and the different syntax.
What is something to focus on your learnings in JavaScript in the next week? |

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -32,6 +32,7 @@
   },
   "dependencies": {
     "core-js": "^2.6.6",
+    "esm": "^3.2.25",
```
Comment (CheezItMan) [inline package.json:35]: Interesting choice!

**Comment by CheezItMan** [comment]
# JS Adagrams
## What We're Looking For



Feature | Feedback
--- | ---
**General** |
Answered comprehension questions |  Check, I'm glad things are going easier.  
Small commits with meaningful commit messages |  More granular commits would be better. 
**Code Requirements** |
`drawLetters` method |   Check
Uses appropriate data structure to store the letter distribution |  Check
All tests for `drawLetters` pass |  Check
`usesAvailableLetters` method |  Check
All tests for `usesAvailableLetters` pass |  Check
`scoreWord` method |  Check
Uses appropriate data structure to store the letter scores |  Check
All tests for `scoreWord` pass |  Check
`highestScoreFrom` method |  Check
Appropriately handles edge cases for tie-breaking logic |  Check
All tests for `highestScoreFrom` pass |  Check
**Overall** |  Nicely done, you even got the class working.  You hit all the learning goals. Congratulations.
